src/solver/type_getters.jl

```diff
@@ -661,6 +687,111 @@ function get_simulation(config::AtmosConfig)
+        @show "DYAMONDSummer, initialize from file"
+        thermo_params = params.thermodynamics_params
+        file_path = AA.dyamond_summer_artifact_path(;
+            context = get_comms_context(config.parsed_args),
+        )
+        p_sfc = Fields.level(
+            ClimaUtilities.SpaceVaryingInputs.SpaceVaryingInput(
+                file_path,
+                "p",
+                spaces.face_space,
+            ),
+            Fields.half,
+        )
+        ᶠT = ClimaUtilities.SpaceVaryingInputs.SpaceVaryingInput(
+            file_path,
+            "t",
+            spaces.face_space,
+        )
+        ᶠq_tot = ClimaUtilities.SpaceVaryingInputs.SpaceVaryingInput(
+            file_path,
+            "q",
+            spaces.face_space,
+        )
+        ᶜT = ClimaUtilities.SpaceVaryingInputs.SpaceVaryingInput(
+            file_path,
+            "t",
+            spaces.center_space,
+        )
+        ᶜq_tot = ClimaUtilities.SpaceVaryingInputs.SpaceVaryingInput(
+            file_path,
+            "q",
+            spaces.center_space,
+        )
+        ᶜp = similar(Y.c.ρ)
+        ᶠ∂lnp∂z = @. -thermo_params.grav / (
+            TD.gas_constant_air(thermo_params, TD.PhasePartition(ᶠq_tot)) * ᶠT
+        )
+        context = get_comms_context(config.parsed_args)
+        for (ᶜp_col, ᶠ∂lnp∂z_col, p_sfc_col) in zip(
+            column_iterator(ᶜp),
+            column_iterator(ᶠ∂lnp∂z),
+            column_iterator(p_sfc),
+        )
+            ClimaCore.Operators.column_integral_indefinite!(
+                ᶜp_col,
+                ᶠ∂lnp∂z_col,
+                log(
+                    ClimaComms.allowscalar(getindex, context.device, p_sfc_col),
+                ),
+            ) # actually computes ln(ᶜp)
+            @. ᶜp_col = exp(ᶜp_col) # replaces ln(ᶜp) with ᶜp
+        end
+        ᶜts = TD.PhaseEquil_pTq.(thermo_params, ᶜp, ᶜT, ᶜq_tot)
+        Y.c.ρ .= TD.air_density.(thermo_params, ᶜts)
+        vel =
+            ClimaCore.Geometry.UVWVector.(
+                ClimaUtilities.SpaceVaryingInputs.SpaceVaryingInput(
+                    file_path,
+                    "u",
+                    spaces.center_space,
+                ),
+                ClimaUtilities.SpaceVaryingInputs.SpaceVaryingInput(
+                    file_path,
+                    "v",
+                    spaces.center_space,
+                ),
+                ClimaUtilities.SpaceVaryingInputs.SpaceVaryingInput(
+                    file_path,
+                    "w",
+                    spaces.center_space,
+                ),
+            )
+        Y.c.uₕ .=
+            ClimaCore.Geometry.Covariant12Vector.(
+                ClimaCore.Geometry.UVVector.(vel)
+            )
+        Y.f.u₃ .=
+            ᶠinterp.(
+                ClimaCore.Geometry.Covariant3Vector.(
+                    ClimaCore.Geometry.WVector.(vel)
+                )
+            )
```

Comment:
I found that it helps with stability if u₃ is not prescribed.

Reply:
This seems to be consistent with @akshaysridhar 's earlier observations !

Reply:
I think it’s because the initial w is too large for our timestep. We can consider adding a threshold too.